packages/@aws-cdk/aws-ec2/lib/security-group.ts

```diff
@@ -149,11 +160,16 @@ export class SecurityGroup extends SecurityGroupRef implements ITaggable {
   private readonly directIngressRules: cloudformation.SecurityGroupResource.IngressProperty[] = [];
   private readonly directEgressRules: cloudformation.SecurityGroupResource.EgressProperty[] = [];
 
+  private readonly allowAllOutbound: boolean;
+
   constructor(parent: Construct, name: string, props: SecurityGroupProps) {
     super(parent, name);
 
     this.tags = new TagManager(this, { initialTags: props.tags});
     const groupDescription = props.description || this.path;
+
+    this.allowAllOutbound = props.allowAllOutbound || false;
```

Comment:
Just for my future coding preference do we prefer:
```
this.allowAllOutbound = props.allowAllOutbound || false;
// or 
this.allowAllOutbound = !!props.allowAllOutbound
```
My bigger question here is defaulting this to `false`. I **think** most users expect this to be `true`? However, other infra code products take this approach so perhaps that is better?

Reply:
My thinking here has been torn between two ways.

This is the "secure by default" solution.

However, it takes more work to get to "what you want", if you want to access things outside your VPC (which you almost always want, to use AWS services). All constructs that create SG's now have to set the default to `true` (which ASGs also do), which is kinda silly.

Yeah, maybe you're right.

Reply:
To be clear I'm also on the fence just like you, just highlighting the pivot in AWS status quo on this particular feature. I can support either direction.

Reply:
Going to go with "works by default" over "secure by default" after all I think.